File: Agent/.NET 6.0/Template.cs
namespace Technologai
{
    public enum TemplateState
    {
        RESTING = 0,
        ASSESSING = 1,
        PROCESSING = 2        
    }
    public class Template 
    {
        public string? Id { get; set; }
        public Data? Description { get; set; }
        public string[]? InputKeys { get; set; }
        public string[]? OutputKeys { get; set; }
        public string? MemberId { get; set; }
        public virtual Task<bool> Assess(Information information) => Task.FromResult(false);
        public virtual Task<Data?> Process(Information information) => Task.FromResult((Data?)null);
    }
}

File: Agent/python_agent/technologai_agent/template.py
from typing import List, Optional
from .information import Information

class Template:
    def __init__(self, id=None, description=None, input_keys=None, output_keys=None, member_id=None):
        self.id = id
        self.description = description
        self.input_keys = input_keys
        self.output_keys = output_keys
        self.member_id = member_id

    async def assess(self, information: Information) -> bool:
        return False

    async def process(self, information: Information) -> Optional[dict]:
        return None